lib/c-api/src/wasm_c_api/module.rs

```diff
@@ -115,3 +157,376 @@ pub unsafe extern "C" fn wasm_module_serialize(
+
+                    const wasm_name_t* export_name = wasm_exporttype_name(export_type);
+                    wasmer_assert_name(export_name, "memory");
+
+                    const wasm_externtype_t* extern_type = wasm_exporttype_type(export_type);
+                    assert(wasm_externtype_kind(extern_type) == WASM_EXTERN_MEMORY);
+
+                    wasm_memorytype_t* memory_type = wasm_externtype_as_memorytype((wasm_externtype_t*) extern_type);
+                    const wasm_limits_t* memory_limits = wasm_memorytype_limits(memory_type);
+                    assert(memory_limits->min == 1);
+                    assert(memory_limits->max == wasm_limits_max_default);
+
+                    wasm_externtype_delete((wasm_externtype_t*) extern_type);
+                }
+
+                wasm_exporttype_vec_delete(&export_types);
+                wasm_byte_vec_delete(wasm);
+                wasm_module_delete(module);
+                wasm_store_delete(store);
+                wasm_engine_delete(engine);
+            }
+        })
+        .success();
+    }
+
+    #[test]
+    fn test_module_imports() {
+        (assert_c! {
+            #include "tests/wasmer_wasm.h"
+
+            int main() {
+                wasm_engine_t* engine = wasm_engine_new();
+                wasm_store_t* store = wasm_store_new(engine);
+
+                wasm_byte_vec_t wat;
+                wasmer_byte_vec_new_from_string(
+                    &wat,
+                    "(module\n"
+                    "  (import \"ns\" \"function\" (func))\n"
+                    "  (import \"ns\" \"global\" (global f32))\n"
+                    "  (import \"ns\" \"table\" (table 1 2 anyfunc))\n"
+                    "  (import \"ns\" \"memory\" (memory 3 4)))"
+                );
+                wasm_byte_vec_t* wasm = wat2wasm(&wat);
+
+                wasm_module_t* module = wasm_module_new(store, wasm);
+                assert(module);
+
+                wasm_importtype_vec_t import_types;
+                wasm_module_imports(module, &import_types);
+
+                assert(import_types.size == 4);
+
+                {
+                    const wasm_importtype_t* import_type = import_types.data[0];
+
+                    const wasm_name_t* import_module = wasm_importtype_module(import_type);
+                    wasmer_assert_name(import_module, "ns");
+
+                    const wasm_name_t* import_name = wasm_importtype_name(import_type);
+                    wasmer_assert_name(import_name, "function");
+
+                    const wasm_externtype_t* extern_type = wasm_importtype_type(import_type);
+                    assert(wasm_externtype_kind(extern_type) == WASM_EXTERN_FUNC);
+
+                    wasm_functype_t* func_type = wasm_externtype_as_functype((wasm_externtype_t*) extern_type);
+
+                    const wasm_valtype_vec_t* func_params = wasm_functype_params(func_type);
+                    assert(func_params->size == 0);
```

Comment:
Optional: consider if you want `assert(func_params && func_params->size == 0);`.

Reply:
Done, thanks!